### DISCUSSION TITLE
Filecoin Name Service Pricing Strategy and Burning Mechanisms

### DISCUSSION THREAD

**Discussion by OpenGate01** [filecoin-project/FIPs#639] [Filecoin Virtual Machine]
Filecoin Name Service

An Filecoin Naming System

 **I. Project Introduction** 
The Filecoin Naming Service (FNS) is a decentralized name service based on the Filecoin EVM. It resolves metadata into easy-to-read xxx.fil names. Each FNS domain is treated as an NFT, and the FNS contract complies with the interfaces of the ERC-721 standard. The NFT can be transferred and traded on secondary markets.

FNS provides a secure and decentralized way to resolve on-chain and off-chain address resources on the Filecoin network into simple human-readable domain names. It is an important part of the Filecoin ecosystem. It can not only make the filecoin system more readable and user-friendly, but also become an important infrastructure for web3.

Users can use FNS as an identity to access the web3 world, and only need to integrate the DNS domain name into FNS, and their domain name can be used as a wallet, Web3 username and a decentralized website.

**2.Pricing Strategy**

1 character：Non-public  registration
2 character：Non-public  registration
3 character：640 USDT/Year
4 character：160 USDT/Year
≥5 character：5 USDT/Year
Ecosystem：Filecoin

**3.Burning Mechanisms**

The first yearly fee exchange to be filecoin and 50% to be burned, 50% for the FNS provider.

**Comment by biquanlibai** [comment]
I think we should use FIL rather than USDT

**Reply by YYTLEO** [reply]
u are right!

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
a fns provider should be open source and non profit

see what https://fns.space/ is doing for example

https://medium.com/@FNS.DAO/announcing-fns-dao-ef2a291d7f5c

a FIP or similar should not include pricing etc anyways but move the ecosystem forward

**Reply by YYTLEO** [reply]
hey there, what do you mean non profit? can you make it free for sure?

**Comment by anorth** [comment]
I don't think a FIP will be required here (any more than ENS would required an EIP), though this discussion is very welcome to spread the idea and gather community feedback.

I will be a bit disappointed if the Filecoin-based naming system that takes hold isn't implemented on the FVM directly as a native actor, using the FRC token standards. Of course bridging to the embedded EVM (and any other guest VMs) would make a lot of sense, but I don't think it makes so much sense for it to actually live there. 

I know we can't do that yet because native actor deployment is not yet enabled. But I'd like to urge developers to target that environment instead.

**Comment by bella000000** [comment]
The project provider may need some profit to have the motivation to move forward.
Paying with fil is indeed a better way to go.
The burn ratio needs to be discussed by the community.

**Comment by sekimlan** [comment]
The market needs to provide a profit as an incentive to attract participation, while having detailed and clear rules on fees will ensure the stability of the market.
What is proposed is very much in need of discussion and meaningful!

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
> The market needs to provide a profit as an incentive to attract participation, while having detailed and clear rules on fees will ensure the stability of the market.

that isn't much of a market then. but fixed pricing.

in a market the pricing will be determined by other factors than hardcoded numbers. hardcoding pricing will only harm adoption

**Reply by sekimlan** [reply]
But there are still issues that need to be discussed here. Without pricing, there can be no competition, so how can a market be formed and how do participants choose different service providers based on what criteria?

**Comment by YYTLEO** [comment]
The economics of naming services is good for the filecoin ecosystem!

**Comment by murphylu666** [comment]
As the ecosystem of filecoin, I think this project can bring more helpful to filecoin. It can promote the relatives applications and show many values of filecoin.

**Comment by Satoshi-cong** [comment]
A healthy ecosystem requires transparent economic models and incentive mechanisms to ensure an open, fair, and just environment that balances the interests of community contributors. Therefore, clear rules are needed on the launch of FVM to enable the healthy development of the community.

**Comment by NikoLuo1010** [comment]
Looks good, keep an eye out for future development

**Comment by YYTLEO** [comment]
The Filecoin Naming Service is an important part of the Filecoin ecosystem, providing a decentralized and user-friendly domain name system for accessing data stored on the Filecoin network. FNS aims to address the difficulties and inconveniences of using complex CIDs to access data, while providing a secure and scalable system that can be easily integrated with other web3 protocols and applications. By treating FNS domains as NFTs, users have greater control and flexibility over their domain ownership and can transfer and trade them on secondary markets. FNS is an important infrastructure for the web3 world, providing a secure and decentralized way for users to access their data and participate in the decentralized internet.

**Comment by YYTLEO** [comment]
come and share your ideas with us  https://discord.com/channels/930392130659160084/1082165232677437512
